pkg/processor/trigger/partitioned/kafka/trigger.go

```diff
@@ -81,3 +85,123 @@ func (k *kafka) CreatePartitions() ([]partitioned.Partition, error) {
 
 	return partitions, nil
 }
+
+func (k *kafka) Stop(force bool) (functionconfig.Checkpoint, error) {
+	offsets := make([]int64, len(k.Partitions))
+
+	for i, abstractPartition := range k.Partitions {
+		kafkaPartition, ok := abstractPartition.(*partition)
+		if !ok {
+			return nil, errors.New("Can't convert partition to kafka partition")
+		}
+		kafkaPartition.Stop()
+		offsets[i] = kafkaPartition.offset
+	}
+
+	out, err := json.Marshal(offsets)
+	if err != nil {
+		k.Logger.ErrorWith("Can't decode offsets to JSON", "err", err)
+		return nil, errors.Wrap(err, "Can't encode offsets to JSON")
+	}
+
+	checkpoint := string(out)
+	return &checkpoint, nil
+}
+
+// AddPartition adds a new partition
+func (k *kafka) AddPartition(partitionConfig *functionconfig.Partition) error {
+	partition, err := k.startPartitionFromConfig(partitionConfig)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create partition")
+	}
+
+	// add partition
+	k.Partitions = append(k.Partitions, partition)
+	return nil
+}
+
+// RemovePartition removes a partition
+func (k *kafka) RemovePartition(partitionConfig *functionconfig.Partition) error {
+	i, kafkaPartition, err := k.findPartition(partitionConfig)
+	if err != nil {
+		return err
+	}
+
+	kafkaPartition.Stop()
```

Comment:
You're losing the offset here and won't be able to recover from where you stopped. One of the key design points is that if you're given a partition to remove you must respond with the offset it was at, no?

Reply:
I thought it's the dealer responsibility to do this book-keeping. I'll talk to @yaronha about this.

Reply:
@tebeka The dealer asks to remove/stop task #N, the processor respond w deleted and checkpoint data (to be used in the init on a new processor)